### ISSUE TITLE
A11Y: Modal component accessibility violations

### ISSUE THREAD

**Issue by alisonemiller** [carbon-design-system/carbon-components-react#599]
## Detailed description

I receive the following accessibility violations against the `Modal` component when I run the DAP plugin.

<img width="1631" alt="screen shot 2018-02-09 at 11 13 14 am" src="https://user-images.githubusercontent.com/17085131/36037362-8ead5970-0d8a-11e8-9948-b7319dd8ac79.png">

> Is this issue related to a specific component?

`Modal`

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

`v5.17.8`

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM Cloud Private
